pytorch_lightning/utilities/distributed.py

```diff
@@ -177,31 +177,6 @@ def sync_ddp(
     return result
 
 
-class AllGatherGrad(torch.autograd.Function):
-    @staticmethod
-    def forward(
-        ctx: Any,
-        tensor: torch.Tensor,
-        group: Optional["torch.distributed.ProcessGroup"] = group.WORLD,
-    ) -> torch.Tensor:
-        ctx.group = group
-
-        gathered_tensor = [torch.zeros_like(tensor) for _ in range(torch.distributed.get_world_size())]
-
-        torch.distributed.all_gather(gathered_tensor, tensor, group=group)
-        gathered_tensor = torch.stack(gathered_tensor, dim=0)
-
-        return gathered_tensor
-
-    @staticmethod
-    def backward(ctx: Any, *grad_output: torch.Tensor) -> Tuple[torch.Tensor, None]:
-        grad_output = torch.cat(grad_output)
-
-        torch.distributed.all_reduce(grad_output, op=torch.distributed.ReduceOp.SUM, async_op=False, group=ctx.group)
-
-        return grad_output[torch.distributed.get_rank()], None
-
-
```

Comment:
this class can't be immediately removed like this as it is part of the stable, public Lightning API. https://pytorch-lightning.readthedocs.io/en/latest/governance.html#api-evolution

Instead, this class should be marked as deprecated in version 1.6 and slated for removal in version 1.8.

However, the implementation of `all_gather_ddp_if_available` can be changed freely to rely on `torch.distributed.all_gather`

Reply:
Thanks for review! Let me check and resend PR :)